File: 351004/Romanovskiy/task361/publisher/src/main/java/com/example/task361/mapper/TweetMapper.java
package com.example.task361.mapper;

import com.example.task361.domain.dto.request.TweetRequestTo;
import com.example.task361.domain.dto.response.TweetResponseTo;
import com.example.task361.domain.entity.Tweet;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping; // ДОБАВИТЬ ЭТО
import org.mapstruct.ReportingPolicy;

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface TweetMapper {
    
    @Mapping(target = "markers", ignore = true) // ДОБАВИТЬ ЭТУ СТРОКУ
    Tweet toEntity(TweetRequestTo request);

    TweetResponseTo toResponse(Tweet tweet);
}

File: 351002/Abrashin/publisher/src/main/kotlin/com/example/distcomp/mapper/TweetMapper.kt
package com.example.distcomp.mapper

import com.example.distcomp.dto.request.TweetRequestTo
import com.example.distcomp.dto.response.TweetResponseTo
import com.example.distcomp.model.Tweet
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.Named

@Mapper(componentModel = "spring", uses = [StickerMapper::class])
interface TweetMapper {
    @Mapping(target = "stickers", source = "stickers", qualifiedByName = ["namesToStickers"])
    fun toEntity(request: TweetRequestTo): Tweet
    
    fun toResponse(entity: Tweet): TweetResponseTo

    @org.mapstruct.Named("namesToStickers")
    fun namesToStickers(names: List<String>?): List<com.example.distcomp.model.Sticker> {
        return names?.map { com.example.distcomp.model.Sticker(name = it) } ?: emptyList()
    }
}
